### PR TITLE
[MONDRIAN-2692] Get non-integer data from snowflake as a BigDecimal;

### PR DESCRIPTION
**Pull request by peterrinehart**
work-around for loss of precision in getDouble method.

### CODE REVIEW

File: mondrian/src/main/java/mondrian/rolap/SqlStatement.java [hunk 1]
```diff
@@ -415,6 +416,20 @@ public Object get() throws SQLException {
...
+        case DECIMAL:
+            return new Accessor() {
+                public Object get() throws SQLException {
+                    final BigDecimal decimal = resultSet.getBigDecimal(columnPlusOne);
+                    if (decimal == null && resultSet.wasNull()) {
+                        return null;
+                    }
+                    final double val = resultSet.getBigDecimal(columnPlusOne).doubleValue();
+                    if ( val == Double.NEGATIVE_INFINITY || val == Double.POSITIVE_INFINITY ) {
+                        throw new ArithmeticException("Big decimal value in " + resultSet.getMetaData().getColumnName(columnPlusOne) + "exceeds double size");
```
Comment (peterrinehart) [inline mondrian/src/main/java/mondrian/rolap/SqlStatement.java:428]: Externalize strings

File: mondrian/src/main/java/mondrian/rolap/SqlStatement.java [hunk 2]
```diff
@@ -415,6 +416,20 @@ public Object get() throws SQLException {
...
             };
+        case DECIMAL:
+            return new Accessor() {
+                public Object get() throws SQLException {
+                    final BigDecimal decimal = resultSet.getBigDecimal(columnPlusOne);
+                    if (decimal == null && resultSet.wasNull()) {
+                        return null;
+                    }
+                    final double val = resultSet.getBigDecimal(columnPlusOne).doubleValue();
+                    if ( val == Double.NEGATIVE_INFINITY || val == Double.POSITIVE_INFINITY ) {
```
Comment (peterrinehart) [inline mondrian/src/main/java/mondrian/rolap/SqlStatement.java:427]: Get spacing rules sorted out.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 30s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

File: mondrian/src/main/java/mondrian/rolap/SqlStatement.java (hunk 2 above)
Comment (peterrinehart) [reply mondrian/src/main/java/mondrian/rolap/SqlStatement.java:435]: Also externalize strings below.

File: mondrian/src/main/java/mondrian/rolap/SqlStatement.java [hunk 3]
```diff
@@ -535,6 +559,7 @@ public Object get(ResultSet resultSet, int column) throws SQLException {
             case LONG:
                 return resultSet.getLong(column + 1);
             case DOUBLE:
+            case DECIMAL:
```
Comment (mkambol) [inline mondrian/src/main/java/mondrian/rolap/SqlStatement.java:562]: I don't remember why we have both Accessors which retrieve resultset values *and* a get() method in TYPE.  But whatever the reason, shouldn't the DECIMAL type use getBigDecimal here?

Comment (peterrinehart) [reply mondrian/src/main/java/mondrian/rolap/SqlStatement.java:562]: Probably.  This seems to be some fallback default that doesn't ever get used, bug we might as well be consistent.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -DrunITs -pl mondrian
```
---
:ok_hand: All tests passed!
 
Tests run: **3705**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/mondrian/job/PR-1229/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 58s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -DrunITs -pl mondrian
```
---
:ok_hand: All tests passed!
 
Tests run: **3705**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/mondrian/job/PR-1229/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 17s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -DrunITs -pl mondrian
```
---
:ok_hand: All tests passed!
 
Tests run: **3705**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/mondrian/job/PR-1229/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: mondrian/src/it/java/mondrian/test/DialectTest.java [hunk 4]
```diff
@@ -1569,6 +1569,22 @@ public void testPostgresGreenplumTypeMapQuirks() throws SQLException {
                 0) == SqlStatement.Type.OBJECT);
     }
 
+    public void testSnowflakeTypeMapQuirks() throws SQLException {
+        MockResultSetMetadata mockResultSetMeta = new MockResultSetMetadata();
+        Dialect greenplumDialect =
+          TestContext.getFakeDialect(Dialect.DatabaseProduct.SNOWFLAKE);
```
Comment (mkambol) [inline mondrian/src/it/java/mondrian/test/DialectTest.java:1575]: greenplum?  Did you copy and paste without updating the var name?

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 24s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -DrunITs -pl mondrian
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by mkambol** [comment]
sonar cleanup is coming in a separate commit

**Pull request merged**